### PR TITLE
fix(tianmu): fix query syntax (WHERE NOT IN ) is not supported. (#767)

### PR DESCRIPTION
**Pull request by lujiashun**
## Summary about this PR

Issue Number: ref #767
[summary]
1 root cause: sub-select item is the left argument,not the right argument of item_func;
2  "in" clause is a liltte different from "not in" clause, This PR just deal with "not in" clause;

## Tests Check List

- [ ] Unit test
- [X] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [X] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: mysql-test/suite/tianmu/t/issue767.test [hunk 1]
```diff
@@ -0,0 +1,7 @@
+--source include/have_tianmu.inc
+use test;
+create table tt(val int) ENGINE=TIANMU;
+insert into tt values (12);
```
Comment (RingsC) [inline mysql-test/suite/tianmu/t/issue767.test]: it's better to adds some dependent and independent subquery cases for `NOT IN`

File: storage/tianmu/core/query.cpp [hunk 2]
```diff
@@ -1667,6 +1668,9 @@ bool Query::ClearSubselectTransformation(common::Operator &oper_for_subselect, I
     if (dynamic_cast<Item_ref_null_helper *>(((Item_func *)cond_removed)->arguments()[1]) == nullptr)
       return false;
     field_for_subselect = nullptr;
+    // the left side of equality should be the left side of the original
+    // expression with subselect
+    left_ref = ((Item_func *)cond_removed)->arguments()[0];
```
Comment (RingsC) [inline storage/tianmu/core/query.cpp]: Using c++ style type casting and `down_cast` would be a good choice.

**Review by RingsC — CHANGES_REQUESTED** [review]

**Comment by lujiashun** [comment]
The sql "in" clause is not support，Its data structure is different from ”not in“ clause. Deal it later.
```
use test;
drop table if exist t1;
create table t1(val int) ENGINE=tianmu;
insert into t1 values(12);
select * from t1 where 42  in (select * from t1 where val > 42);
```

**Review by RingsC — APPROVED** [review]
LGTM, Great job.

**Review by hustjieke — APPROVED** [review]
LGTM

**Pull request merged**